presto-main/src/main/java/io/prestosql/sql/analyzer/Field.java

```diff
@@ -169,4 +171,26 @@ public String toString()
 
         return result.toString();
     }
+
+    public static class Source
```

Comment:
`SourceRelation`? or `OriginRelation` because you have `getOrigin`.

I don't like `Source` because it is very ambiguous.